llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -690,6 +690,13 @@ void SelectionDAGLegalize::LegalizeLoadOps(SDNode *Node) {
       assert(NVT.getSizeInBits() == VT.getSizeInBits() &&
              "Can only promote loads to same size type");
 
+      // If the range metadata type does not match the legalized memory
+      // operation type, remove the range metadata.
+      if (const MDNode *MD = LD->getRanges()) {
+        ConstantInt *Lower = mdconst::extract<ConstantInt>(MD->getOperand(0));
+        if (Lower->getBitWidth() != NVT.getScalarSizeInBits())
```

Comment:
Should also drop if the element type isn't integer

Reply:
Invalidate range MD if legalized type is not integer.